batchflow/models/eager_torch/base.py

```diff
@@ -0,0 +1,1203 @@
+            inputs = self._fill_value(inputs)
+        return inputs
+
+    def _fill_input(self, *args, **kwargs):
+        if args and kwargs:
+            raise ValueError('Use either positional or keyword arguments in `train` call.')
+        if kwargs:
+            for name in ['labels', 'masks', 'targets']:
+                if name in kwargs:
+                    targets = kwargs.pop(name)
+            args = [kwargs.get(name) for name in (self.input_names or list(kwargs.keys()))]
+            args.append(targets)
+        return tuple([self._fill_param(arg) for arg in args])
+
+    def _fill_output(self, fetches, outputs):
+        _fetches = [fetches] if isinstance(fetches, str) else fetches
+
+        output = []
+        for f in _fetches:
+            if f in outputs:
+                v = outputs[f]
+                if isinstance(v, (torch.Tensor, torch.autograd.Variable)):
+                    v = v.detach().cpu().numpy()
+                output.append(v)
+            else:
+                raise KeyError('Unknown value to fetch', f)
+
+        output = output[0] if isinstance(fetches, str) else type(fetches)(output)
+        return output
+
+
+    def train(self, *args, feed_dict=None, fetches=None, use_lock=False, train_mode='',
+              accumulate_grads=True, sync_frequency=True, microbatch=None, **kwargs):
+        """ Train the model with the data provided
+
+        Parameters
+        ----------
+        args
+            Arguments to be passed directly into the model.
+        fetches : tuple, list
+            Sequence of tensor names to calculate and return.
+        use_lock : bool
+            If True, the whole train step is locked, thus allowing for multithreading.
+        train_mode : str or sequence of str
+            Name(s) of train step(s) to optimize. Regular expressions are allowed.
+            If multiple train steps are selected (either via passing a sequence or by using regular expression),
+            then all of them are optimized sequentially.
+        accumulate_grads : bool
+            If True, then gradients from different train modes are accumulated and applied once at the end.
+            If False, then gradients are applied for each of the train modes separately.
+        sync_frequency : int, bool or None
+            If int, then how often to apply accumulated gradients to the weights.
+            If True, then value from config is used (default value is to apply gradients after each batch of data).
+            If False or None, then gradients are applied after each batch of data.
+        microbatch : int, bool or None
+            If int, then size of chunks to split every batch into. Allows to process given data sequentially,
+            accumulating gradients from microbatches and applying them once in the end.
+            If True, then value from config is used (default value is not to use microbatching).
+            If False or None, then microbatching is not used.
```

Comment:
Please update docstring for `microbatch == None` - is value from config used or not?